src/test/proxy/worker_proxy.py

```diff
@@ -0,0 +1,253 @@
+            if uid == job.uid:
+                try:
+                    job.status = JobStatus.RUNNING
+                    return WorkerResponse.from_dict(WorkerResponse(
+                        result_string=IWorkerProxy.RESUME_JOB_SUCCESS % (uid, self.uid), is_success=True).to_dict())
+                except ValueError:
+                    return WorkerResponse.from_dict(WorkerResponse(
+                        result_string=IWorkerProxy.RESUME_JOB_NOT_PAUSED % (uid, self.uid), is_success=False
+                    ).to_dict())
+        return WorkerResponse.from_dict(WorkerResponse(
+            result_string=IWorkerProxy.RESUME_JOB_UNKNOWN_JOB % (uid, self.uid), is_success=False).to_dict())
+
+
+class AbstractWorkerProxyTest(TestCase):
+    def setUp(self) -> None:
+        self._uid_1 = "job-1"
+        self._job_1 = Job(
+            owner_id=1003,
+            email=None,
+            scheduling_constraints=JobSchedulingConstraints(
+                priority=JobPriority.MEDIUM, is_preemtible=True, special_resources=[]
+            ),
+            docker_context=DockerContext(dockerfile_source="", mount_points=[]),
+            docker_constraints=DockerConstraints(cpu_threads=1, memory=1024)
+        )
+        self._job_1.uid = self._uid_1
+        self._job_1.status = JobStatus.QUEUED
+        self._uid_2 = "job-2"
+        self._job_2 = Job(
+            owner_id=1003,
+            email=None,
+            scheduling_constraints=JobSchedulingConstraints(
+                priority=JobPriority.MEDIUM, is_preemtible=True, special_resources=[]
+            ),
+            docker_context=DockerContext(dockerfile_source="", mount_points=[]),
+            docker_constraints=DockerConstraints(cpu_threads=1, memory=1024)
+        )
+        self._job_2.uid = self._uid_2
+        self._job_2.status = JobStatus.QUEUED
+        self._uid_unknown = "DEADBEEF"
+        self._empty_worker: IWorkerProxy = None
+        self._busy_worker: IWorkerProxy = None
+
+    @skipIfAbstract
+    def test_dispatch_to_empty_worker(self) -> None:
+        response_1 = self._empty_worker.dispatch_job(self._job_1)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._empty_worker.dispatch_job(self._job_2)
+        self.assertTrue(response_2.is_success)
+
+    @skipIfAbstract
+    def dispatch_to_busy_worker(self) -> None:
+        response_1 = self._busy_worker.dispatch_job(self._job_1)
+        self.assertFalse(response_1.is_success)
+        response_2 = self._busy_worker.dispatch_job(self._job_2)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_dispatch_same_job_twice_1(self) -> None:
+        response_1 = self._empty_worker.dispatch_job(self._job_1)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._empty_worker.dispatch_job(self._job_1)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_dispatch_same_job_twice_2(self) -> None:
```

Comment:
this is a duplicate test, I don't think it makes much sense to have it. At least put it in a for loop in the same test ;)

Reply:
This is not a duplicate test because I'm assuring that the position of the job in the list does not affect the test results.

Reply:
Sorry, I misrememberd how I arranged the jobs.
It's true that this test does not add much value but at the same time it took almost no effort to create.

Reply:
It is very unlikely that the actual job used makes a difference but I think with unit tests you should test even things that you're relatively certain about.

Reply:
I understand it is easy to add more tests like this, but I think you should also consider what would happen if we need to tweak the interface a little bit - you'd need to make the same changes in a very repetitive manner to fix the tests, hence I suggested grouping tests together with for-loops.